applications/ShapeOptimizationApplication/python_scripts/mesh_controllers/mesh_controller_with_solver.py

```diff
@@ -162,13 +164,15 @@ def UpdateMeshAccordingInputVariable(self, variable):
         self.OptimizationModelPart.ProcessInfo.SetValue(KM.TIME, time_before_update)
         self.OptimizationModelPart.ProcessInfo.SetValue(KM.DELTA_TIME, delta_time_before_update)
 
+        self.skip_first_remshing_step = False
+
         KM.Logger.PrintInfo("ShapeOpt", "Time needed for updating the mesh = ",round(timer.time() - startTime,2),"s")
 
     # --------------------------------------------------------------------------
     def Finalize(self):
         self._mesh_moving_analysis.Finalize()
 
-        if self.is_remeshing_used:
+        if self.is_remeshing_used and not self.skip_first_remshing_step:
```

Comment:
as far as i understand this will always be called, since you are setting the variable to false in line 167?

Reply:
Not exactly, it is a safe guard against if someone forgets to call `UpdateMeshAccordingInputVariable`, then I don't want to call `Finalize` in the meshing process.